Add API for Articles and Resources

srri-zz opened the pull request:
Addition of:
- API endpoint for `articles` and `resources` models
- Added new requirement for mysqldb (python)
- Changed some imports to reflect the new structure

srri committed:
initial addition of articles, resources

srri committed:
add routing

srri committed:
add new requirement

srri-zz commented:
@psiemens or @blx, thoughts?

psiemens commented on apps/api/serializers.py:
```diff
@@ -1,12 +1,24 @@
 __author__ = 'Steven Richards'
 from django.contrib.auth.models import Group
 from django.contrib.auth import get_user_model
+from dispatch.apps.content.models import Resource, Article
```
I believe this import statement can be simplified like so:

`from content.models import Resource, Article`

Django autoloads all the apps so you don't have to include the `dispatch.apps` part of the import statement, just the app name. Give it a try and see if it works.

blx replied:
On the other hand, we could keep the fully qualified name to be explicit about what `content` or `core` etc are.

psiemens replied:
Yeah, that might be best. Good to avoid naming clashes with external libs

psiemens commented:
I don't think we need an API route for Resource because it's just a base model. Am I right, @blx?

The mysqldb requirement is a good call.

blx commented:
Yeah it's abstract, I'm not sure it needs to be exposed in the API. As long as `Article` properly inherits the `Resource` fields.

Should `django` itself be in requirements? (If I make a new virtualenv and just install djangorestframework it doesn't pull down django.)

psiemens commented:
Yeah, I guess it should be.

srri committed:
added requirement and removed resource API route

srri-zz commented:
Fixed the route for resource, I wasn't sure what is was in terms of our modelling, thanks for clarifying. Also added django as a requirement. Merging since I've made those changes
